inc/import/parser/class.ParserFITMultiple.php

```diff
@@ -11,101 +11,78 @@
 /**
  * Abstract parser for multiple activities in *.fit-file
  *
- * @author Hannes Christiansen
+ * @author undertrained
  * @package Runalyze\Import\Parser
  */
 class ParserFITMultiple extends ParserAbstractMultiple {
-	/** @var string */
-	const PERL_FIT_ERROR_MESSAGE_START = 'main::Garmin::FIT';
-
-	/** @var string */
-	const PERL_GENERAL_MESSAGE_START = 'perl: warning:';
-
 	/**
-	 * Name of output file
-	 * @var string
+	 * Fit Data
+	 * @var adriangibbons\phpFITFileAnalysis
 	 */
-	protected $Filename = '';
+	protected $fitData = null;
 
 	/**
-	 * Handle
-	 * @var resource
+	 * Set fit data
+	 * @param object $fit
 	 */
-	protected $Handle = null;
-
-	/**
-	 * Set filename
-	 * @param string $filename
-	 */
-	public function setFilename($filename) {
-		$this->Filename = $filename;
+	public function setFitData($fit) {
+		$this->fitData = $fit;
 	}
 
 	/**
 	 * Parse
 	 */
 	public function parse() {
-		$firstActivityStarted = false;
-		$SingleParser = new ParserFITSingle('');
+		if (isset($this->fitData->data_mesgs['activity']['num_sessions']) &&
+		    $this->fitData->data_mesgs['activity']['num_sessions'] > 1) {
+			for ($session = 0; $session < $this->fitData->data_mesgs['activity']['num_sessions']; $session++) {
+				$fitsingle = clone $this->fitData;
```

Comment:
I did not take a deep look into the structure of that fitData object, but is this cloning really required? As the fitData already holds the complete fit file with large associative arrays, this may require a lot of ram. (But I did not compare time and ram consumption of this new parser versus the perl one so far)